go/arrow/util/protobuf_reflect.go

```diff
@@ -0,0 +1,447 @@
+package util
+
+import (
+	"fmt"
+	"github.com/apache/arrow/go/v16/arrow"
+	"github.com/apache/arrow/go/v16/arrow/array"
+	"github.com/apache/arrow/go/v16/arrow/memory"
+	"github.com/huandu/xstrings"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/known/anypb"
+	"reflect"
+)
+
+type SchemaOptions struct {
+	exclusionPolicy    func(pfr ProtobufFieldReflection) bool
+	fieldNameFormatter func(str string) string
+}
```

Comment:
since this has no exported members or methods and we don't seem to return this or take it as a parameter anywhere, should we just change this so it's not exported anymore?